src/marine/smos_sss2ioda.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python
+
+#
+# (C) Copyright 2019 UCAR
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+#
+
+import sys
+import argparse
+import numpy as np
+from datetime import datetime, timedelta
+import netCDF4 as nc
+from pathlib import Path
+
+IODA_CONV_PATH = Path(__file__).parent/"../lib/pyiodaconv"
+if not IODA_CONV_PATH.is_dir():
+    IODA_CONV_PATH = Path(__file__).parent/'..'/'lib-python'
+sys.path.append(str(IODA_CONV_PATH.resolve()))
+
+import ioda_conv_ncio as iconv
+from orddicts import DefaultOrderedDict
+
+
+vName = "sea_surface_salinity"
+
+locationKeyList = [
+    ("latitude", "float"),
+    ("longitude", "float"),
+    ("datetime", "string")
+]
+
+AttrData = {
+    'odb_version': 1,
+}
+
+
+class Salinity(object):
+    def __init__(self, filenames, date, writer):
+        self.filenames = filenames
+        self.date = date
+        self.data = DefaultOrderedDict(lambda: DefaultOrderedDict(dict))
+        self.writer = writer
+        self._read()
+
+    # Open obs file and read/load relevant info
+    def _read(self):
+        valKey = vName, self.writer.OvalName()
+        errKey = vName, self.writer.OerrName()
+        qcKey = vName, self.writer.OqcName()
+        
+        for f in self.filenames:
+            print(" Reading file: ",f)
+	    ncd = nc.Dataset(f, 'r')
+            
+            source_var_name = {
+                 'lat': 'Latitude',
+		 'lon': 'Longitude',
+                 'sss': 'SSS_corr',
+		 'sss_err':'Sigma_SSS_corr',
+                 'sss_qc': 'Dg_quality_SSS_corr',
+                }
+            lon = ncd.variables['Longitude'][:]
+            lat = ncd.variables['Latitude'][:]
+            sss = ncd.variables['SSS_corr'][:]
+            sss_err = ncd.variables['Sigma_SSS_corr'][:]
+            sss_qc = ncd.variables['Dg_quality_SSS_corr'][:]
+            sss_qc = sss_qc.astype(int)
+
+	    mask = np.logical_not(sss.mask)
+            lon = lon[mask]
+            lat = lat[mask]
+            sss = sss[mask]
+            sss_err = sss_err[mask]
+            sss_qc = sss_qc[mask]
+
+            for i in range(len(lon)):
+                if sss_qc[i] <= 150:
+                   sss_qc[i] = 0
+                else:
+                   sss_qc[i] = 1
+                # get date from filename
+                I = f.find("SM_")
```

Comment:
not to pick on this, but you are using lowercase `i` and uppercase `I` in the same loop.